run_task.py

```diff
@@ -337,17 +368,25 @@ def run(self):
 		self.workspace = folders[0]
 		self.tasks = []
 
-		project_file = OSUtils.find_file_with_pattern(self.workspace, PROJECT_FILE_NAME_PATTERN)
+		project_file = self.window.project_file_name()
+
 		if project_file:
 			with open(project_file, "r") as fp:
+
 				try:
-					tasks_json = json.load(fp)
+					json_str = fp.read()
+					# sublime supports trailing commas in the json files, json doesn't. Fix this here
+					
+					pattern = r',\s*([\]}])'
+					json_str_fixed = re.sub(pattern, r'\1', json_str)
+					tasks_json = json.loads(json_str_fixed)
```

Comment:
There seems to be a much simpler and cleaner solution to this project file parsing issue. We could take advantage of the [`sublime.decode_value(data: str)`](https://www.sublimetext.com/docs/api_reference.html#sublime.decode_value) function and refactor your code to this:
```python
json_str = fp.read()
tasks_json = sublime.decode_value(json_str)
```
As an added bonus, the above code will also gracefully handle the comments in the project file, as per the Sublime API Reference.